tests/e2e/upgrade.go

```diff
@@ -0,0 +1,192 @@
+package e2e
+
+import (
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap.com/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/controller"
+	"github.com/pingcap/tidb-operator/pkg/label"
+	apps "k8s.io/api/apps/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func testUpgrade() {
+	By("Then start upgrade to new version")
+	err := wait.Poll(5*time.Second, 5*time.Minute, upgrade)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Then members should upgrade by order: pd ==> tikv ==> tidb")
+	err = wait.Poll(5*time.Second, 5*time.Minute, memberUpgraded)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Then all members should running")
+	err = wait.Poll(5*time.Second, 5*time.Minute, allMembersRunning)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Then the data is correct")
+	err = wait.Poll(5*time.Second, 5*time.Minute, dataIsCorrect)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func upgrade() (bool, error) {
+	tc, err := cli.PingcapV1alpha1().TidbClusters(ns).Get(clusterName, metav1.GetOptions{})
+	if err != nil {
+		logf("failed to upgrade tidbcluster, error: %v", err)
+		return false, err
+	}
+
+	tc.Spec.PD.Image = strings.Replace(tc.Spec.PD.Image, getImageTag(tc.Spec.PD.Image), "v2.0.0", -1)
```

Comment:
Do we have TiDB `v2.0.0` docker images in our CI? It's better to define version as a constant.

Reply:
I have pushed v2.0.0 to registry manually.